Ginger/Ginger/RunSetPageLib/NewRunSetPage.xaml.cs

```diff
@@ -1918,82 +1919,97 @@ private async void xRunRunsetBtn_Click(object sender, RoutedEventArgs e)
-        //                return;
-        //            }
-        //            UpdateReRunFailedButtonIcon(true);
-
-        //            ResetALMDefectsSuggestions();
-
-
-        //            //run analyzer
-        //            if (mRunSetConfig.RunWithAnalyzer)
-        //            {
-        //                int analyzeRes = await AnalyzeRunsetWithUI().ConfigureAwait(false);
-        //                if (analyzeRes == 1)
-        //                {
-        //                    return;//cancel run because issues found
-        //                }
-        //            }
-
-        //            var result = await WorkSpace.Instance.RunsetExecutor.RunRunsetAsync().ConfigureAwait(false);
-
-        //            // handling ALM Defects Opening
-
-        //            if (WorkSpace.Instance.RunsetExecutor.DefectSuggestionsList != null && WorkSpace.Instance.RunsetExecutor.DefectSuggestionsList.Count > 0)
-        //            {
-        //                ObservableList<ALMDefectProfile> ALMDefectProfiles = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ALMDefectProfile>();
-        //                if (ALMDefectProfiles != null && ALMDefectProfiles.Count > 0)
-        //                {
-        //                    this.Dispatcher.Invoke(() =>
-        //                    {
-        //                        InitALMDefectsOpeningSection();
-        //                    });
-        //                }
-
-        //            }
-        //        }
-        //        catch (Exception ex)
-        //        {
-        //            Reporter.ToLog(eLogLevel.ERROR, "Runset execution failed: ", ex);
-        //        }
-        //        finally
-        //        {
-        //            UpdateReRunFailedButtonIcon();
-        //        }
-        //    }
-        //    else
-        //    {
-        //        return;
-        //    }
-        //}
+                    if (runnerNames.Any())
+                    {
+                        Reporter.ToUser(eUserMsgKey.StaticInfoMessage, $"{string.Join(", ", runnerNames)} is empty, please add {GingerDicser.GetTermResValue(eTermResKey.BusinessFlows)} to run.");
+                        return;
+                    }
+                    UpdateReRunFailedButtonIcon();
+
+                    ResetALMDefectsSuggestions();
+
+
+                    //run analyzer
+                    if (mRunSetConfig.RunWithAnalyzer)
+                    {
+                        int analyzeRes = await AnalyzeRunsetWithUI().ConfigureAwait(false);
+                        if (analyzeRes == 1)
+                        {
+                            return;//cancel run because issues found
+                        }
+                    }
+
+                    var result = await WorkSpace.Instance.RunsetExecutor.RunRunsetAsync().ConfigureAwait(false);
+
+                    // handling ALM Defects Opening
+                    foreach(BusinessFlow bf in DeactivatedBfList)
```

Comment:
Please add this part in finally method to make sure it will executed in case of exception

Reply:
Done